File: companion/spss/python/b04.py
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt

d = pd.read_csv("companion/spss/csv/b04.csv")
assert not d.isna().any().any()
print(d[["ort5", "ort30"]].agg(
    ["count", "mean", "std", "min", "max"]))
fig, axes = plt.subplots(1, 2)
for ax, v in zip(axes, ["ort5", "ort30"]):
    x = d[v]
    ax.hist(x, bins=np.arange(0, 20.5, .5), density=True)
    grid = np.linspace(0, 20, 1000)
    ax.plot(grid, stats.norm.pdf(grid, x.mean(), x.std()))
    ax.set_title(v)
    ax.set_xlim(0, 20); ax.set_ylim(0, .6)
plt.tight_layout(); plt.show()

File: companion/spss/r/b04.R
d <- read.csv("companion/spss/csv/b04.csv")
stopifnot(!anyNA(d))
print(sapply(d[c("ort5", "ort30")], function(v)
  c(n=length(v), mean=mean(v), sd=sd(v),
    min=min(v), max=max(v))))
par(mfrow=c(1, 2))
for (v in c("ort5", "ort30")) {
  x <- d[[v]]
  hist(x, breaks=seq(0, 20, by=.5),
       probability=TRUE, main=v, xlab=v,
       xlim=c(0, 20), ylim=c(0, .6))
  curve(dnorm(x, mean(d[[v]]), sd(d[[v]])),
        add=TRUE, col="blue")
}
par(mfrow=c(1, 1))